File: app/src/main/java/com/iutcalendar/alarm/condition/AlarmConditionRecycleView.java
package com.iutcalendar.alarm.condition;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckedTextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.calendar.iutcalendar.R;
import com.iutcalendar.alarm.AlarmRing;
import com.iutcalendar.alarm.ClickForUpdateListener;
import com.iutcalendar.alarm.condition.label_constraint.AlarmLabelConstraintRecycleView;
import com.iutcalendar.alarm.condition.label_constraint.AlarmConstraintLabel;
import com.iutcalendar.calendrier.DateCalendrier;

import java.util.GregorianCalendar;
import java.util.List;

public class AlarmConditionRecycleView extends RecyclerView.Adapter<AlarmConditionViewHolder> {

    List<AlarmCondtion> list;
//    AlarmConditionViewHolder viewHolder;
    Context context;
    ClickForUpdateListener updateListener, reloadListener;

    public AlarmConditionRecycleView(Context context, List<AlarmCondtion> list, ClickForUpdateListener updateListener, ClickForUpdateListener reload) {
        this.list = list;
        this.context = context;
        this.updateListener = updateListener;
        this.reloadListener = reload;
    }

    @NonNull
    @Override
    public AlarmConditionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);

        View eventView = inflater.inflate(R.layout.condition_alarm_card, parent, false);

        return new AlarmConditionViewHolder(eventView);
//        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull AlarmConditionViewHolder holder, int position) {
        AlarmCondtion alarmConstraint = list.get(position);

        holder.begingHour.setText(DateCalendrier.timeLongToString(alarmConstraint.getBeging()));
        holder.endHour.setText(DateCalendrier.timeLongToString(alarmConstraint.getEnd()));
        holder.ringHour.setText(DateCalendrier.timeLongToString(alarmConstraint.getAlarmAt()));


        holder.begingHour.setOnClickListener(view -> {
            AlarmRing.askTime(context, (view1, hourOfDay, minute) -> {
                alarmConstraint.setBeging(DateCalendrier.getHourInMillis(hourOfDay, minute));
                updateListener.update();
            });
        });
        holder.endHour.setOnClickListener(view -> {
            AlarmRing.askTime(context, (view1, hourOfDay, minute) -> {
                alarmConstraint.setEnd(DateCalendrier.getHourInMillis(hourOfDay, minute));
                updateListener.update();
            });
        });
        holder.ringHour.setOnClickListener(view -> {
            AlarmRing.askTime(context, (view1, hourOfDay, minute) -> {
                alarmConstraint.setAlarmAt(DateCalendrier.getHourInMillis(hourOfDay, minute));
                updateListener.update();
            });
        });

        holder.delBtn.setOnClickListener(view -> {
            AlarmConditionManager.getInstance(context).removeCondition(position);
            reloadListener.update();
        });


        //setCheckable days
        initDayCheck(alarmConstraint, holder.monday, GregorianCalendar.MONDAY);
        initDayCheck(alarmConstraint, holder.tuesday, GregorianCalendar.TUESDAY);
        initDayCheck(alarmConstraint, holder.wednesday, GregorianCalendar.WEDNESDAY);
        initDayCheck(alarmConstraint, holder.thursday, GregorianCalendar.THURSDAY);
        initDayCheck(alarmConstraint, holder.friday, GregorianCalendar.FRIDAY);
        initDayCheck(alarmConstraint, holder.saturday, GregorianCalendar.SATURDAY);
        initDayCheck(alarmConstraint, holder.sunday, GregorianCalendar.SUNDAY);
    }

    private void initDayCheck(AlarmCondtion alarmConstraint, CheckedTextView check, int value) {
        if (alarmConstraint.getDaysEnabled().contains(value)) {
            check.setChecked(true);
        }

        check.setOnClickListener(v -> {
            check.toggle();

            if (check.isChecked())
                alarmConstraint.getDaysEnabled().add(value);
            else
                alarmConstraint.getDaysEnabled().remove(Integer.valueOf(value));

            updateListener.update();
        });
    }


    @Override
    public int getItemCount() {
        return list.size();
    }
}

File: app/src/main/java/com/iutcalendar/alarm/condition/AlarmConditionRecycleView.kt
package com.iutcalendar.alarm.condition

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.CheckedTextView
import android.widget.TimePicker
import androidx.recyclerview.widget.RecyclerView
import com.iutcalendar.alarm.AlarmRing
import com.iutcalendar.calendrier.DateCalendrier
import com.univlyon1.tools.agenda.R
import java.util.*

class AlarmConditionRecycleView(
    var context: Context?,
    var list: List<AlarmCondition>?,
    private var updateItemListener: (itemPos: Int) -> Unit,
    private var removeItemListener: (itemPos: Int) -> Unit,
    private var saveListener: () -> Unit
) : RecyclerView.Adapter<AlarmConditionViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AlarmConditionViewHolder {
        val context = parent.context
        val inflater = LayoutInflater.from(context)
        val eventView = inflater.inflate(R.layout.condition_alarm_card, parent, false)
        return AlarmConditionViewHolder(eventView)
    }

    override fun onBindViewHolder(holder: AlarmConditionViewHolder, position: Int) {
        val alarmConstraint = list!![position]
        holder.beginHour.text = DateCalendrier.timeLongToString(alarmConstraint.begin)
        holder.endHour.text = DateCalendrier.timeLongToString(alarmConstraint.end)
        holder.ringHour.text = DateCalendrier.timeLongToString(alarmConstraint.alarmAt)
        holder.beginHour.setOnClickListener {
            AlarmRing.askTime(context) { _: TimePicker?, hourOfDay: Int, minute: Int ->
                alarmConstraint.begin = DateCalendrier.getHourInMillis(hourOfDay, minute)
                updateItemListener(position)
            }
        }
        holder.endHour.setOnClickListener {
            AlarmRing.askTime(context) { _: TimePicker?, hourOfDay: Int, minute: Int ->
                alarmConstraint.end = DateCalendrier.getHourInMillis(hourOfDay, minute)
                updateItemListener(position)
            }
        }
        holder.ringHour.setOnClickListener {
            AlarmRing.askTime(context) { _: TimePicker?, hourOfDay: Int, minute: Int ->
                alarmConstraint.alarmAt = DateCalendrier.getHourInMillis(hourOfDay, minute)
                updateItemListener(position)
            }
        }
        holder.delBtn.setOnClickListener {
            AlarmConditionManager.getInstance(context).removeCondition(position)
            removeItemListener(position)
        }


        //setCheckable days
        initDayCheck(alarmConstraint, holder.monday, GregorianCalendar.MONDAY)
        initDayCheck(alarmConstraint, holder.tuesday, GregorianCalendar.TUESDAY)
        initDayCheck(alarmConstraint, holder.wednesday, GregorianCalendar.WEDNESDAY)
        initDayCheck(alarmConstraint, holder.thursday, GregorianCalendar.THURSDAY)
        initDayCheck(alarmConstraint, holder.friday, GregorianCalendar.FRIDAY)
        initDayCheck(alarmConstraint, holder.saturday, GregorianCalendar.SATURDAY)
        initDayCheck(alarmConstraint, holder.sunday, GregorianCalendar.SUNDAY)
    }

    private fun initDayCheck(alarmConstraint: AlarmCondition, check: CheckedTextView?, value: Int) {
        if (alarmConstraint.daysEnabled.contains(value)) {
            check!!.isChecked = true
        }
        check!!.setOnClickListener {
            check.toggle()
            if (check.isChecked) alarmConstraint.daysEnabled.add(value)
            else alarmConstraint.daysEnabled.remove(Integer.valueOf(value))

            saveListener()
        }
    }

    override fun getItemCount(): Int {
        return list!!.size
    }
}